From acc01dbd373f8820374bf50b7384f2d5673884c7 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Thu, 30 Aug 2012 13:52:19 +0800
Subject: [PATCH 332/635] ENGR00221752  No Alarm output for HDMI

Alarm will set two device, HDMI, Speaker. HAL need to select HDMI for
output in this case.

Signed-off-by: b02247 <b02247@freescale.com>
---
 alsa/tinyalsa_hal.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/alsa/tinyalsa_hal.c b/alsa/tinyalsa_hal.c
index 84729f3..67b442c 100644
--- a/alsa/tinyalsa_hal.c
+++ b/alsa/tinyalsa_hal.c
@@ -398,7 +398,13 @@ static int start_output_stream(struct imx_stream_out *out)
     /* S/PDIF takes priority over HDMI audio. In the case of multiple
      * devices, this will cause use of S/PDIF or HDMI only */
     for(i = 0; i < MAX_AUDIO_CARD_NUM; i++) {
-        if((out->device & AUDIO_DEVICE_OUT_ALL) & adev->card_list[i]->supported_devices) {
+        if(out->device & AUDIO_DEVICE_OUT_AUX_DIGITAL) {
+            if(adev->card_list[i]->supported_devices & AUDIO_DEVICE_OUT_AUX_DIGITAL) {
+                   card = adev->card_list[i]->card;
+                   port = 0;
+                   break;
+            }
+        } else if((out->device & AUDIO_DEVICE_OUT_ALL) & adev->card_list[i]->supported_devices) {
             card = adev->card_list[i]->card;
             port = 0;
             break;
-- 
1.8.0

